packages/hmr-plugin/src/symbols.ts

```diff
@@ -0,0 +1,31 @@
+import { StateOperations } from '@ngxs/store';
+import { NgModuleRef } from '@angular/core';
+
+export interface NgxsStoreSnapshot<T = any> {
+  [key: string]: T | T[];
+}
+
+export enum NGXS_HMR {
+  hmrNgxsStoreOnInit = 'hmrNgxsStoreOnInit',
+  hmrNgxsStoreBeforeOnDestroy = 'hmrNgxsStoreBeforeOnDestroy',
+  SNAPSHOT_KEY = '__NGXS_HMR_SNAPSHOT__'
+}
+
+export interface NgxsHmrLifeCycle<T = NgxsStoreSnapshot> {
+  /**
+   * hmrNgxsStoreOnInit is called when the AppModule on init
+   * @param context - current state from Store
+   * @param snapshot - previous state from Store after last hmr on destroy
+   */
+  hmrNgxsStoreOnInit(context: StateOperations<T>, snapshot: Partial<T>): void;
+
+  /**
+   * hmrNgxsStoreOnInit is called when the AppModule on destroy
+   * @param context - current state from Store
+   */
+  hmrNgxsStoreBeforeOnDestroy(context: StateOperations<T>): T;
+}
+
+export type HmrNgxsStoreOnInitFn<T> = (context: StateOperations<T>, snapshot: Partial<T>) => void;
```

Comment:
Did you know that you can alternatively refer to this signature like this:
```TS
const hmrNgxsStoreOnInitFn: NgxsHmrLifeCycle<T>['hmrNgxsStoreOnInit'] = ref.instance[NGXS_HMR.hmrNgxsStoreOnInit];
```
So, you could avoid having to expost this signature again and just use the interface method signature that is already exposed.
... same for the on destroy method.

Reply:
Fixed in my recent commit to this branch.